### PR TITLE
Fixing broken link in documentation

### LINKED ISSUE #3744: Edit on GitHub link issues
### Description

The edit on GitHub link from the docs is broken for a number of versions.
### Expected behavior

Click the link and be brought to the code on GitHub
### Actual behavior

Error as link does not exist (sometimes)
### Steps to Reproduce

Pick you favourite documentation (e.g https://docs.sunpy.org/en/stable/api/sunpy.net.fido_factory.UnifiedResponse.html) click on on the 'edit on github' link. Change the version using the version box in the bottom right of the webpage.

| doc version  | link | status |
| ------------ | ----|------- |
| latest  | https://github.com/sunpy/sunpy/tree/master/sunpy/net/fido_factory.py#L31  | ok |
| stable  | https://github.com/sunpy/sunpy/tree/1.1.0/sunpy/net/fido_factory.py#L31 | broken |
| v1.1.0 | https://github.com/sunpy/sunpy/tree/1.1.0/sunpy/net/fido_factory.py#L31 | broken |
| v1.0.7 | https://github.com/sunpy/sunpy/tree/1.0.7/sunpy/net/fido_factory.py#L31 | broken |
| v0.9.9 | http://github.com/sunpy/sunpy/tree/0.9/sunpy/net/fido_factory.py#L30 | ok |
| v0.8.5 | http://github.com/sunpy/sunpy/tree/v0.8.x/sunpy/net/fido_factory.py#L29 | broken |

For `1.x.y` the url needs to be `.../tree/v1.x.y/...` but for the the `v0.8.5` the x in the url is the issue so `.../tree/0.8.5/...` works but only without the leading  'v'?

It's not a major issue just an annoyance especially for for the 1.1.0 docs.

Seem's like this line is the issue but not sure how to fix for both 1.x.y and 0.x.y maybe add an if?
https://github.com/sunpy/sunpy/blob/fe8ef2ec505d504cb2af52d90246529e8c5fb84f/docs/conf.py#L242

### PR DESCRIPTION
**Pull request by crusader2000**
### Description

Fixes #3744

### CODE REVIEW

**Comment by nabobalis** [comment]
Thanks for the PR @crusader2000.

Unforutaly, I don't understand the rationale behind this change and how it fixes that issue.

File: docs/conf.py [hunk 1]
```diff
@@ -239,7 +239,10 @@ def rstjinja(app, docname, source):
 # "version" configuration parameter
 edit_on_github_project = "sunpy/sunpy"
 if 'dev' not in release:
-    edit_on_github_branch = f"{version}"
+	edit_on_github_branch = f"{version}"
+	if version[0]=='0':
```
Comment (dstansby) [inline docs/conf.py]: Could you add a comment in the code above this line to say what it is doing? I'm a bit confused.

File: changelog/3828.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+- Fixed the issue of broken link. Details of the issue can be found at #3744. (`#3744 https://github.com/sunpy/sunpy/pull/3828`__)
```
Comment (dstansby) [inline changelog/3828.bugfix.rst]:
```suggestion
Fixed broken "edit on github" links in the documentation.
```
You don't have to link to the issue explicitly, and I've suggested a more descriptive sentence.

**Comment by nabobalis** [comment]
Sorry but I am gonna close this since I have a fix in #3917

**Pull request closed without merging**